File: Test/AdventureWorksFunctionalModel/Person/BusinessEntityAddress.cs
namespace AW.Types;

    [Named("Address")]
    public class BusinessEntityAddress : IHasRowGuid, IHasModifiedDate {
        [Hidden]
        public int BusinessEntityID { get; init; }

        [MemberOrder(3)]
        public virtual BusinessEntity BusinessEntity { get; init; }


        [Hidden]
        public int AddressTypeID { get; init; }

        [MemberOrder(1)]
        public virtual AddressType AddressType { get; init; }

        [Hidden]
        public int AddressID { get; init; }

        [MemberOrder(2)]

        public virtual Address Address { get; init; }

        
        [MemberOrder(99),Versioned]
        public DateTime ModifiedDate { get; init; }

        [Hidden]
        public Guid rowguid { get; init; }

        public override string ToString() => $"{AddressType}: {Address}";
    }

File: Test/AdventureWorksLegacyModelVB/Person/BusinessEntityAddress.vb
Namespace AW.Types

	'<Named("Address")>
	Partial Public Class BusinessEntityAddress

		Implements ITitledObject

		Public Property BusinessEntityID() As Integer

		<MemberOrder(3)>
		Public Overridable Property BusinessEntity() As BusinessEntity

		Public Property AddressTypeID() As Integer

		<MemberOrder(1)>
		Public Overridable Property AddressType() As AddressType

		'<Hidden>
		Public Property AddressID() As Integer

		<MemberOrder(2)>
		Public Overridable Property Address() As Address

#Region "ModifiedDate"
		Public Property mappedModifiedDate As Date
		Friend myModifiedDate As TimeStamp

		<MemberOrder(99)>
		Public ReadOnly Property ModifiedDate As TimeStamp
			Get
				Return If(myModifiedDate, New TimeStamp(mappedModifiedDate, Function(v) mappedModifiedDate = v))
			End Get
		End Property

		Public Sub AboutModifiedDate(a As FieldAbout)
			Select Case a.TypeCode
				Case AboutTypeCodes.Usable
					a.Usable = False
			End Select
		End Sub
#End Region

		Public Property RowGuid() As Guid

		Public Function Title() As ITitle Implements ITitledObject.Title
			Return New Title(ToString())
		End Function

		Public Overrides Function ToString() As String
			Return $"{AddressType}: {Address}"
		End Function
	End Class
End Namespace